app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -294,3 +312,391 @@ def test_hanging_protocol_clientside():
+                "secondary": ["test-ci-image-1", "test-ci-overlay-1"],
+                "tertiary": ["test-ci-image-2", "test-ci-overlay-2"],
+                "quaternary": ["test-ci-image-3", "test-ci-overlay-3"],
+                "quinary": ["test-ci-image-4", "test-ci-overlay-4"],
+            },
+        ),
+        (
+            1,
+            6,
+            3,
+            {
+                "main": ["test-ci-isolated-0"],
+                "secondary": ["test-ci-image-0", "test-ci-overlay-0"],
+                "tertiary": ["test-ci-image-1", "test-ci-overlay-1"],
+                "quaternary": ["test-ci-image-2", "test-ci-overlay-2"],
+                "quinary": ["test-ci-image-3"],
+                "senary": ["test-ci-image-4"],
+                "septenary": ["test-ci-image-5"],
+            },
+        ),
+        (
+            1,
+            3,
+            6,
+            {
+                "main": ["test-ci-isolated-0"],
+                "secondary": [
+                    "test-ci-image-0",
+                    "test-ci-overlay-0",
+                    "test-ci-overlay-1",
+                ],
+                "tertiary": [
+                    "test-ci-image-1",
+                    "test-ci-overlay-2",
+                    "test-ci-overlay-3",
+                ],
+                "quaternary": [
+                    "test-ci-image-2",
+                    "test-ci-overlay-4",
+                    "test-ci-overlay-5",
+                ],
+            },
+        ),
+        (
+            1,
+            3,
+            8,
+            {
+                "main": ["test-ci-isolated-0"],
+                "secondary": [
+                    "test-ci-image-0",
+                    "test-ci-overlay-0",
+                    "test-ci-overlay-1",
+                    "test-ci-overlay-2",
+                ],
+                "tertiary": [
+                    "test-ci-image-1",
+                    "test-ci-overlay-3",
+                    "test-ci-overlay-4",
+                    "test-ci-overlay-5",
+                ],
+                "quaternary": [
+                    "test-ci-image-2",
+                    "test-ci-overlay-6",
+                    "test-ci-overlay-7",
+                ],
+            },
+        ),
+    ),
+)
+def test_generate_view_content_example(
+    number_of_isolated_ci,
+    number_of_images,
+    number_of_overlays,
+    expected_example_json,
+):
+    for i in range(number_of_isolated_ci):
+        ComponentInterfaceFactory(
+            kind=InterfaceKindChoices.CHART, title=f"test-ci-isolated-{i}"
+        )
+
+    for i in range(number_of_images):
+        ComponentInterfaceFactory(
+            kind=InterfaceKindChoices.IMAGE, title=f"test-ci-image-{i}"
+        )
+
+    for i in range(number_of_overlays):
+        ComponentInterfaceFactory(
+            kind=InterfaceKindChoices.SEGMENTATION,
+            title=f"test-ci-overlay-{i}",
+        )
```

Comment:
Can't you use the `ci_list` fixture here as well?

Reply:
Upon synchronous discussion, a helper function is made and used here as well.
Done.